plugins/bufferlines/bufferline.nix

```diff
@@ -1,320 +1,435 @@
+          in
+          optionSnakeCase
+        ) (oldHighlightOptions ++ [ "truncMarker" ]);
+      in
+      genAttrs newHighlightsOptions (
         name:
-        helpers.mkNullOrOption helpers.nixvimTypes.highlight ''
+        defaultNullOpts.mkHighlight { } null ''
           Highlight group definition for ${name}.
         ''
       );
-    };
   };
 
-  config =
-    let
-      setupOptions = with cfg; {
-        options = {
-          inherit mode themable numbers;
-          buffer_close_icon = bufferCloseIcon;
-          modified_icon = modifiedIcon;
-          close_icon = closeIcon;
-          close_command = closeCommand;
-          left_mouse_command = leftMouseCommand;
-          right_mouse_command = rightMouseCommand;
-          middle_mouse_command = middleMouseCommand;
-          inherit indicator;
-          left_trunc_marker = leftTruncMarker;
-          right_trunc_marker = rightTruncMarker;
-          separator_style = separatorStyle;
-          name_formatter = nameFormatter;
-          truncate_names = truncateNames;
-          tab_size = tabSize;
-          max_name_length = maxNameLength;
-          color_icons = colorIcons;
-          show_buffer_icons = showBufferIcons;
-          show_buffer_close_icons = showBufferCloseIcons;
-          get_element_icon = getElementIcon;
-          show_close_icon = showCloseIcon;
-          show_tab_indicators = showTabIndicators;
-          show_duplicate_prefix = showDuplicatePrefix;
-          enforce_regular_tabs = enforceRegularTabs;
-          always_show_bufferline = alwaysShowBufferline;
-          persist_buffer_sort = persistBufferSort;
-          max_prefix_length = maxPrefixLength;
-          sort_by = sortBy;
-          inherit diagnostics;
-          diagnostics_indicator = diagnosticsIndicator;
-          diagnostics_update_in_insert = diagnosticsUpdateInInsert;
-          inherit offsets;
-          groups = {
-            inherit (groups) items;
-            options = {
-              toggle_hidden_on_enter = groups.options.toggleHiddenOnEnter;
-            };
-          };
-          hover = {
-            inherit (hover) enabled reveal delay;
-          };
-          debug = {
-            inherit (debug) logging;
-          };
-          custom_filter = customFilter;
-        } // cfg.extraOptions;
-
-        highlights = mapAttrs (
-          pluginOptionName: nixvimOptionName: cfg.highlights.${nixvimOptionName}
-        ) highlightOptions;
-      };
-    in
-    mkIf cfg.enable {
-      extraPlugins = with pkgs.vimPlugins; [
-        cfg.package
-        nvim-web-devicons
-      ];
-      opts.termguicolors = true;
-      extraConfigLua = ''
-        require('bufferline').setup${helpers.toLuaObject setupOptions}
+  callSetup = false;
+  extraConfig = cfg: {
+    extraConfigLua =
+      let
+        setupOptions = {
+          options = cfg.settings;
+
+          inherit (cfg) highlights;
+        };
```

Comment:
We wouldn't normally do this; the `settings` option should literally be "options provided to `setup`".

In other words, we should have `settings.options` and `settings.highlights`.

Reply:
@MattSturgeon I tried setting up the renames to go under the settings.options and settings.highlights, but am getting `error: invalid regular expression '.*vim.loader.disable().*'` running the bufferline test. Not sure if you can spot something I'm missing here.

Reply:
Well, I was getting that on my macbook... not getting it on my desktop..

Reply:
> @MattSturgeon I tried setting up the renames to go under the settings.options and settings.highlights, but am getting `error: invalid regular expression '.*vim.loader.disable().*'` running the bufferline test. Not sure if you can spot something I'm missing here.

Interesting, we'd need the `--show-trace` output to dig into that a bit more. Have you pushed the change that caused the issue?

At first glance, it looks like the string `"vim.loader.disable()"` is being converted into a regex without being properly escaped. Though the actual regex looks to just be re-implementing `lib.hasInfix` anyway.

Reply:
Not sure what was going on with it, but after rebooting I can run the tests on my macbook again.